src/main/kotlin/com/exactpro/th2/codec/xml/XmlPipelineCodec.kt

```diff
@@ -19,56 +19,74 @@ import com.exactpro.th2.codec.DecodeException
 import com.exactpro.th2.codec.api.IPipelineCodec
 import com.exactpro.th2.codec.xml.utils.toMap
 import com.exactpro.th2.codec.xml.utils.toProto
+import com.exactpro.th2.codec.xml.utils.toTransport
 import com.exactpro.th2.codec.xml.xsd.XsdValidator
-import com.exactpro.th2.common.grpc.AnyMessage
-import com.exactpro.th2.common.grpc.Message
-import com.exactpro.th2.common.grpc.MessageGroup
-import com.exactpro.th2.common.grpc.RawMessage
-import com.exactpro.th2.common.message.logId
-import com.exactpro.th2.common.message.messageType
-import com.exactpro.th2.common.message.toJson
-import com.github.underscore.lodash.Xml
+import com.exactpro.th2.common.message.*
+import com.exactpro.th2.common.grpc.AnyMessage as ProtoAnyMessage
+import com.exactpro.th2.common.grpc.Message as ProtoMessage
+import com.exactpro.th2.common.grpc.MessageGroup as ProtoMessageGroup
+import com.exactpro.th2.common.grpc.RawMessage as ProtoRawMessage
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.MessageGroup
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.ParsedMessage
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.RawMessage
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.toByteArray
+import com.github.underscore.Xml
 import com.google.protobuf.ByteString
-import org.slf4j.Logger
-import org.slf4j.LoggerFactory
+import io.netty.buffer.Unpooled
+import mu.KotlinLogging
 import java.nio.charset.Charset
 import java.nio.file.Path
+import java.util.*
 
 open class XmlPipelineCodec(private val settings: XmlPipelineCodecSettings, xsdMap: Map<String, Path>)  : IPipelineCodec {
 
     private val pointer = settings.typePointer?.split("/")?.filterNot { it.isBlank() }
     private var xmlCharset: Charset = Charsets.UTF_8
     private val validator = XsdValidator(xsdMap, settings.dirtyValidation)
 
-    override fun encode(messageGroup: MessageGroup): MessageGroup {
+    override fun encode(messageGroup: ProtoMessageGroup): ProtoMessageGroup {
```

Comment:
Please, migrate to `fun ...(messageGroup: MessageGroup, context: IReportingContext)` decode/encode methods